src/xbot/common/properties/DatabaseStorageBase.java

```diff
@@ -0,0 +1,173 @@
+            log.warn(e.toString());
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public boolean obliterateStorage() {
+        try {
+            Connection conn = DriverManager.getConnection(dbUrl);
+
+            if (propertiesTableExists(conn)) {
+                Statement sta = conn.createStatement();
+                String payload = "DROP TABLE PROPERTIES";
+
+                int response = sta.executeUpdate(payload);
+
+                if (response == 0) {
+                    return true;
+                }
+                // something went wrong
+                return false;
+            } else {
+                // table does not exist, nothing to obliterate
+                return true;
+            }
+        } catch (SQLException e) {
+            log.warn("Something went wrong when attempting to drop the PROPERTIES table!");
+            log.warn(e.toString());
+            e.printStackTrace();
+            return false;
+        }
+
+    }
+
+    protected String readFromFile() {
+        // Need to read out the database into a string
+
+        try {
+            Connection conn = DriverManager.getConnection(dbUrl);
+
+            if (propertiesTableExists(conn)) {
+                Statement sta = conn.createStatement();
+                String payload = "SELECT * FROM PROPERTIES";
+
+                ResultSet rs = sta.executeQuery(payload);
+
+                StringBuffer sb = new StringBuffer();
+
+                while (rs.next()) {
+                    sb.append(rs.getString("Type"));
+                    sb.append(propertyDelimiter);
+                    sb.append(rs.getString("Name"));
+                    sb.append(propertyDelimiter);
+                    sb.append(rs.getString("Value"));
+                    sb.append(lineSeperator);
+                }
+
+                return sb.toString();
+            } else {
+                // could not load properties table.
+                log.error("Could not find the properties table in the database when loading properties!");
+                log.error("All properties will be at their default levels!");
+            }
+        } catch (SQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        return "";
+    }
+
+    protected void writeToFile(String data) {
+
+        Connection conn = null;
+
+        try {
+            // we need to be more resilient here, and only create if table doesn't exist.
+            conn = DriverManager.getConnection(dbUrl);
+        } catch (SQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+
+        try {
+            // create table if it doesn't exist
+            Statement sta = conn.createStatement();
+            String payload = "CREATE TABLE PROPERTIES (Name VARCHAR(100), Type VARCHAR(20), Value VARCHAR(50))";
+            int count = sta.executeUpdate(payload);
```

Comment:
If we're not validating that we create the database with the count variable, shouldn't we remove it, as we're not using it?